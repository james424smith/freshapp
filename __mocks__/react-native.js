import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
