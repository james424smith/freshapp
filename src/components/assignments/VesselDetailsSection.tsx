import React from "react";
import { View, Image } from "react-native";
import styles from "../../styles/AssignmentScreenStyles";
import Text from "../StyledText";
import { VesselDetails } from "../../interfaces";
import { Theme, useTheme } from "@react-navigation/native";
type Props = {
  vesselDetails?: VesselDetails;
};

const renderVesselDetails = (theme: Theme, vesselDetails?: VesselDetails) => {
  return (
    <View>
      <View style={styles.vesselDetails}>
        <View style={styles.columnsContainer}>
          <View>
            <Text
              style={{
                ...styles.vesselDetailsLabel,
                color: theme.colors.text,
              }}
            >
              {"Vessel Name"}
            </Text>
            <Text
              style={{
                ...styles.vesselDetailsValue,
                color: theme.colors.text,
              }}
            >
              {vesselDetails?.name}
            </Text>
          </View>
        </View>
        <View style={styles.columnsContainer}>
          <View style={styles.firstContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"IMO"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.vesselImoNumber}
            </Text>
          </View>

          <View style={styles.secondContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Flag"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.flag}
            </Text>
          </View>
        </View>
        <View style={styles.columnsContainer}>
          <View style={styles.firstContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"GT"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.grossTonnage}
            </Text>
          </View>

          <View style={styles.secondContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"DWT"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.deadWeightTonnage}
            </Text>
          </View>
        </View>
        <View style={styles.columnsContainer}>
          <View style={styles.firstContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Draught"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.draught}
            </Text>
          </View>

          <View style={styles.secondContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Beam"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.beam}
            </Text>
          </View>
        </View>
        <View style={styles.columnsContainer}>
          <View style={styles.firstContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Length o.a."}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.lengthOverall}
            </Text>
          </View>

          <View style={styles.secondContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Length b.p."}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.lengthBetweenPerpendiculars}
            </Text>
          </View>
        </View>
      </View>
      <View style={styles.lineStyle} />
      <View style={styles.vesselDetails}>
        <View style={styles.columnsContainer}>
          <View>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Main engine series"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.mainEngineSeries}
            </Text>
          </View>
        </View>
        <View style={styles.columnsContainer}>
          <View style={styles.firstContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Main engine marker"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.mainEngineMaker}
            </Text>
          </View>

          <View style={styles.secondContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"Main engine output"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.mainEngineOutput}
            </Text>
          </View>
        </View>
        <View style={styles.columnsContainer}>
          <View style={styles.firstContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"ECDIS Manufacturer"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.ecdisManufacture}
            </Text>
          </View>

          <View style={styles.secondContainerWidth}>
            <Text
              style={{ ...styles.vesselDetailsLabel, color: theme.colors.text }}
            >
              {"ECDIS Model"}
            </Text>
            <Text
              style={{ ...styles.vesselDetailsValue, color: theme.colors.text }}
            >
              {vesselDetails?.ecdisModel}
            </Text>
          </View>
        </View>
      </View>
    </View>
  );
};

const VesselDetailsSection = (props: Props) => {
  const { vesselDetails } = props;
  const theme = useTheme();
  const imageUri =
    vesselDetails?.vesselImage &&
    `data:image/png;base64,${vesselDetails?.vesselImage}`;

  return (
    <View
      style={{
        ...styles.outerContainer,
        backgroundColor: theme.colors.background,
      }}
    >
      <Image
        style={styles.vesselImageStyle}
        source={{
          uri: imageUri,
        }}
      />
      <View style={styles.row}>
        <Text style={{ ...styles.vesselTitle, color: theme.colors.text }}>
          Vessel
        </Text>
        <Text style={styles.vesselTitle} />
      </View>
      <View style={styles.vesselDetailsContainer}>
        {renderVesselDetails(theme, vesselDetails)}
      </View>
    </View>
  );
};

export default VesselDetailsSection;
