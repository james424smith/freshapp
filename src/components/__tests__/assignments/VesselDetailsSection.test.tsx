import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import VesselDetailsSection from "../../assignments/VesselDetailsSection";
import { vesselDetails } from "../../../../__mocks__/fake-data/assignmentDetails.json";

describe("VesselDetailsSection snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render VesselDetailsSection correctly as per the ui made when vesselImage is defined", () => {
    const { toJSON } = render(
      <VesselDetailsSection vesselDetails={vesselDetails} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render VesselDetailsSection correctly as per the ui made when vesselImage is not defined", () => {
    const { toJSON } = render(
      <VesselDetailsSection
        vesselDetails={{ ...vesselDetails, vesselImage: undefined }}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
