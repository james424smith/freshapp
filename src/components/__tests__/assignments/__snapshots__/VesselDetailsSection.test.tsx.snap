// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VesselDetailsSection snapshot test should render VesselDetailsSection correctly as per the ui made when vesselImage is defined 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "data:image/png;base64,1234",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 347,
        "width": undefined,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 34,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "rgb(28, 28, 30)",
            "flexBasis": "40%",
            "fontSize": 46.5,
            "fontWeight": "bold",
            "marginHorizontal": 30,
          },
        ]
      }
    >
      Vessel
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "#2A2E43",
            "flexBasis": "40%",
            "fontSize": 46.5,
            "fontWeight": "bold",
            "marginHorizontal": 30,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 75,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexBasis": "40%",
            "margin": 30,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Vessel Name
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              WEC BRUEGHEL
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              IMO
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              9143972
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Flag
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              PMD
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              GT
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              6362
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              DWT
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              7224
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Draught
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              6.69
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Beam
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              18.45
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Length o.a.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              121.76
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Length b.p.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              114.9
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#78849E",
            "borderBottomWidth": 0.3,
            "marginBottom": 45,
            "marginLeft": 37.5,
            "marginTop": 22.5,
          }
        }
      />
      <View
        style={
          Object {
            "flexBasis": "40%",
            "margin": 30,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Main engine series
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              L40/54
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Main engine marker
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              MAN B&W
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Main engine output
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              5300
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              ECDIS Manufacturer
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              ECDIS Model
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`VesselDetailsSection snapshot test should render VesselDetailsSection correctly as per the ui made when vesselImage is not defined 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 347,
        "width": undefined,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 34,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "rgb(28, 28, 30)",
            "flexBasis": "40%",
            "fontSize": 46.5,
            "fontWeight": "bold",
            "marginHorizontal": 30,
          },
        ]
      }
    >
      Vessel
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "#2A2E43",
            "flexBasis": "40%",
            "fontSize": 46.5,
            "fontWeight": "bold",
            "marginHorizontal": 30,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 75,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexBasis": "40%",
            "margin": 30,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Vessel Name
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              WEC BRUEGHEL
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              IMO
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              9143972
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Flag
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              PMD
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              GT
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              6362
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              DWT
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              7224
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Draught
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              6.69
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Beam
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              18.45
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Length o.a.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              121.76
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Length b.p.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              114.9
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#78849E",
            "borderBottomWidth": 0.3,
            "marginBottom": 45,
            "marginLeft": 37.5,
            "marginTop": 22.5,
          }
        }
      />
      <View
        style={
          Object {
            "flexBasis": "40%",
            "margin": 30,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Main engine series
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              L40/54
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Main engine marker
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              MAN B&W
            </Text>
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              Main engine output
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              5300
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 41.5,
            }
          }
        >
          <View
            style={
              Object {
                "width": "55%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              ECDIS Manufacturer
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "width": "45%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                  },
                ]
              }
            >
              ECDIS Model
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
